cli/root.go

```diff
@@ -356,6 +358,13 @@ func (r *RootCmd) Command(subcommands []*clibase.Cmd) (*clibase.Cmd, error) {
 			Value:       clibase.StringArrayOf(&r.header),
 			Group:       globalGroup,
 		},
+		{
+			Flag:        varHeaderProcess,
+			Env:         "CODER_HEADER_PROCESS",
+			Description: "An external process that outputs JSON-encoded key-value pairs to be used as additional HTTP headers added to all requests.",
```

Comment:
The main reason was because originally this header command thing was being ran by the VS Code plugin and JSON felt natural for TypeScript, but I think you are right, newline-delimited sounds way easier to script.  Will change!